server/app/views/admin/questions/MapQuestionSettingsFilterFragment.html

```diff
@@ -0,0 +1,47 @@
+<th:block
+  th:fragment="filterInput(index, possibleKeys, selectedKey, displayNameValue)"
+>
+  <div class="display-flex flex-justify filter-input">
+    <div>
+      <label
+        class="usa-label"
+        th:for="'filters[' + ${index} + '].key'"
+        th:text="#{map.keyLabel}"
+      ></label>
+      <select
+        class="usa-select"
+        th:name="'filters[' + ${index} + '].key'"
+        th:id="'filters[' + ${index} + '].key'"
```

Comment:
For `name` this is correct to do arrays.

For `id` you should only use alphanumerics, hyphens, and underscores. HTML is more lenient, but javascript is more fickle and at best will require escaping some special characters to use them as IDs. It's easier to just keep the id value simpler.